tanjun/checks.py

```diff
@@ -787,3 +788,63 @@ def with_check(check: tanjun_abc.CheckSig, /) -> collections.Callable[[CommandT]
         A command decorator callback which adds the check.
     """
     return lambda command: command.add_check(check)
+
+
+def with_any_role_check(
+    roles: list[str | int] = list(),
+    *,
+    error_message: str | None = "You do not have the required roles to use this command!",
+    halt_execution: bool = True,
+) -> collections.Callable[[CommandT], CommandT]:
+    """Only let a command run if the author has a specific role.
+
+    Parameters
+    ----------
+    roles: list[str | int]
+        The author must have at least one (1) role in this list. (Role.name and Role.id are checked)
+
+    Other Parameters
+    ----------------
+    error_message: str | None
+        The error message raised if the member does not have a required role.
+
+        Defaults to 'You do not have the required roles to use this command!'
+    halt_execution: bool
+        Whether this check should raise `tanjun.errors.HaltExecution` to
+        end the execution search when it fails instead of returning `False`.
+
+        Defaults to `False`.
+
+    Returns
+    -------
+    collections.abc.Callable[[CommandT], CommandT]
+        A command decorator callback which adds the check."""
+
+    def any_role_check(
+        ctx: tanjun_abc.Context,
+        *,
+        roles: list[str],
+        error_message: str | None = None,
+        halt_execution: bool = False,
+    ) -> bool:
+
+        if not ctx.member:
+            return _handle_result(False, "You must be a member to use this!", True)
+
+        member_roles = ctx.member.get_roles()
```

Comment:
It would probably make sense to fallback to fetching the roles from rest if necessary to make this check more compatible with a cacheless bot (e.g. interaction server).

Plus if we process the roles while initialising the class for this check we could add a hot path for when no role names are provided which avoids getting the role objects all together and rather just checks against ctx.member.role_ids